lib/kernel/src/prim_tty.erl

```diff
@@ -1359,6 +1359,8 @@ insert_buf(State, Bin, LineAcc, Acc) ->
                            State#state.buffer_expand =:= undefined ->
             [PrevChar | BB] = State#state.buffer_before,
             case string:next_grapheme([PrevChar | Bin]) of
+                [$\e | _] ->
```

Comment:
Can we add a comment here about what is going on? Something like:

Ansi escape sequences can never have unicode characters in them, so if PrivChar is such a value we know that this char is not part of that cluster.